app/streamSelector/browser.js

```diff
@@ -1,13 +1,41 @@
 const { desktopCapturer, ipcRenderer } = require('electron');
-window.addEventListener('DOMContentLoaded', createSourceSelector());
+window.addEventListener('DOMContentLoaded', init());
 
-function createSourceSelector() {
+function init() {
 	return () => {
-		ipcRenderer.once('get-screensizes-response', (event, screens) => {
-			createPreviewScreen(screens);
-		});
-		ipcRenderer.send('get-screensizes-request');
-	};
+		if (process.env["XDG_SESSION_TYPE"] === "wayland")
+			initRequestSource();
+		else
+			createSourceSelector();
+	}
+}
+
+function initRequestSource() {
```

Comment:
this function and the one in line 34 can use the callback pattern (to remove the duplicated code).

I will add suggestions that should work (but I can't currently test)